src/Compiler.Dynamic/SystemWebCompilation.cs

```diff
@@ -110,56 +111,51 @@ private async Task<T> InternalCompilePageAsync(string path, CancellationToken to
     {
         try
         {
-            var queue = new Queue<string>();
-            queue.Enqueue(path);
-
+            List<string> allSortedFiles = GetAllDependencies(path);
```

Comment:
why? once you override the correct methods in the usercontrolparser it should light up and handle the queueing of items